### PR TITLE
Set PVC name on VolumeGroupSnapshot members

### PR DESCRIPTION
**Pull request by leonardoce**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This patch sets the `spec.persistentVolumeClaimName` field for
volume snapshots that are members of a dynamically provisioned volume
group snapshot, like a dynamically provisioned volume snapshot would do.

The volume snapshot logic has been improved to skip the creation of
relative snapshot as it has already been created by the group snapshot.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
The "spec.persistentVolumeClaimName" field is now set on VolumeSnapshots that are members of a dynamically provisioned VolumeGroupSnapshot. This is consistent with dynamically provisioned VolumeSnapshots.
```

### CODE REVIEW

File: pkg/common-controller/snapshot_controller.go [hunk 1]
```diff
@@ -484,6 +484,40 @@ func (ctrl *csiSnapshotCommonController) syncUnreadySnapshot(snapshot *crdv1.Vol
...
+	if _, ok := snapshot.Labels[utils.VolumeGroupSnapshotNameLabel]; ok {
+		if snapshot.Status == nil || snapshot.Status.BoundVolumeSnapshotContentName == nil {
+			klog.V(5).Infof(
+				"syncUnreadySnapshot [%s]: detected group snapshot member with no content, retrying",
+				utils.SnapshotKey(snapshot))
+			return fmt.Errorf("detected group snapshot member %s with no content, retrying",
+				utils.SnapshotKey(snapshot))
+		}
+
+		volumeGroupContentName := *snapshot.Status.BoundVolumeSnapshotContentName
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_controller.go:497]: volumeGroupContentName -> volumeSnapshotContentName

File: pkg/common-controller/groupsnapshot_controller_helper.go [hunk 2]
```diff
@@ -621,7 +621,7 @@ func (ctrl *csiSnapshotCommonController) createSnapshotsForGroupSnapshotContent(
 				"createSnapshotsForGroupSnapshotContent: creating volumesnapshotcontent %w", err)
 		}
 
-		_, err = ctrl.clientset.SnapshotV1().VolumeSnapshots(volumeSnapshotNamespace).Create(ctx, volumeSnapshot, metav1.CreateOptions{})
+		createdVolumeSnapshot, err := ctrl.clientset.SnapshotV1().VolumeSnapshots(volumeSnapshotNamespace).Create(ctx, volumeSnapshot, metav1.CreateOptions{})
```
[the remark this answers is not in the document]
Comment (leonardoce) [reply pkg/common-controller/groupsnapshot_controller_helper.go:624]: I would have loved to do that.

Unfortunately [it is not possible](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource). That's why I'm forced to hit the API server two times.

**Comment by yati1998** [comment]
@leonardoce were you able to test creating Volumegroupsnapshot with your changes?

**Pull request merged**